src/@providers/docker/services/deployment.ts

```diff
@@ -172,6 +154,49 @@ export class DockerDeploymentService extends CrudResourceService<'deployment', D
       throw new Error(stderr || 'Deployment failed');
     }
 
+    // Deployment successful. Report back to the server.
+    const inspectionRes = await this.inspect(containerName);
+    const networks = Object.keys(inspectionRes?.NetworkSettings.Networks || {});
+    const ipAddress = inspectionRes?.NetworkSettings.Networks[networks[0]].IPAddress;
+
+    for (const serviceConfig of inputs.services || []) {
+      const account = this.providerStore.getProvider(serviceConfig.account);
+      if (!account) {
+        throw new Error(`Cannot be registered w/ service. Account does not exist: ${serviceConfig.account}`);
+      }
+
+      const service = account.resources.service;
+      if (!service) {
+        throw new Error(
+          `Cannot be registered w/ service. ${serviceConfig.account} does not support service resources.`,
+        );
+      }
+
+      if (!('construct' in service) && !('create' in service)) {
+        throw new Error(`The ${account.type} provider cannot create service resources`);
+      }
+
+      const writeableService = service as unknown as WritableResourceService<'service', DockerCredentials>;
```

Comment:
Can we make some sort of helper function that can handle this instead of putting so much non-docker specific code into this class?

Reply:
That's a weakness that already exists simply because the providerStore is injected into the providers/services. I agree with you, but let's discuss it outside the scope of this PR so we can move forward.

Reply:
https://github.com/architect-team/arcctl/issues/110

Reply:
Fixed.